moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -108,8 +144,76 @@ export default function MapView({ projectName, projectStatuses }) {
   };
 
   const onSaveComponent = () => {
-    const newComponents = [...components, draftComponent];
-    setComponents(newComponents);
+    /* Start data preparation */
```

Comment:
I thought about extracting these data prep lines into a helper, but I kind of like seeing the data built up for insertion. Down to refactor if there are strong opinions about it.

Reply:
My inclination is to not break things out into a helper until they are called multiple times and you're doing it to DRY up your code. This, as written, is the process for building up the data for insertion, and it's going to be a complex operation no matter where the code steps reside -- at least like this, you can see it all at once! This is actually a straight-forward, step-by-step set of instructions for the task, as written. I think you've nailed it.

(This is a weakly held assertion about utility / helper collections - it's not been too long since I wrote the VZ ETL and I pulled a ton of stuff into a helper when it was only called once -- because it gave a big boost to readability. Aways more than one way to do it!)